content/specs/xrpc.md

```diff
@@ -78,6 +78,30 @@ The set of endpoints requiring admin auth is likely to get out of date in this s
 - `com.atproto.server.createInviteCode`
 - `com.atproto.server.createInviteCodes`
 
+### Inter-Service Authentication (Temporary Specification)
+
+This section describes a mechanism for authentication between services using signed JWTs. Note that this is not the long-term solution for inter-service authentication, and is likely to be replaced with a more robust mechanism in the near future.
+
+The current mechanism is to use short-lived JWTs signed by the account's atproto signing key. The receiving service can validate the signature by checking this key against the account's DID document.
+
+The JWT parameters are:
+
+- `alg` header field: indicates the signing key type (see [Cryptography](/specs/cryptography))
+    - use `ES256K` for `k256` keys
+    - use `ES256` for `p256` keys
+- `iss` body field: account DID that the request is being sent on behalf of
+- `aud` body field: service DID associated with the service that the request is being sent to
+- `exp` body field: token expiration time, as a UNIX timestamp with seconds precision. Should be a short time window, as revocation is not implemented. 60 seconds is a good value.
+- JWT signature: base64url-encoded signature using the account DID's signing key
+
+The signature is computed using the regular JWT process. As Typescript pseudo-code, this looks like:
+
+```
+const headerPayload = utf8ToBase64Url(jsonStringify(header)) + '.' + utf8ToBase64Url(jsonString(body))
+const signature = sign(sha256(utf8Bytes(headerPayload)))
```

Comment:
my bad on not including it in the pseudo code - but maybe clarify in this function that `sign` is done with the user's atproto signing key - same as a repo signature

Reply:
Sort of related, I personally think of the `sha256()` as being part of the signing process.  At the same time I appreciate the explicitness here about how to generate the signature.  I think the risk would be someone double-hashing by mistake, not realizing that their keypair might be doing that on its own, so if there's a way to make a clarification to avoid that it could be useful.  But also not a huge deal and probably clear enough for most implementers, not terribly worried about this.

Reply:
My feeling on this is that in our SDKs we should call the primary function `hash_and_sign()`. I think it is equally confusing to have a convention that `sign()` also implicitly hashes.

A python-ic alternative would be something like `sign(raw_data, hash="sha256")` as an (optional) keyword arg.

Reply:
yeah I agree with these & we probably shouldn't explicilty say `sha256` here

I'd be game for saying `hashAndSign` in the pseudo-code & we can look at adjusting it in the SDK as well

I've found there is annoyingly not a strong convention in crypto libs around if the hash is part of the signing process :/

Reply:
The pair of `hashAndSign` is also `hashAndVerify`, which sounds even worse, but has also bitten me a bunch as well. Have thought about these a bunch in the context of having relatively consistent SDK API shape across programming languages.